### ISSUE TITLE
Field group is hidden if all fields are inside fieldsets

### ISSUE THREAD

**Issue by rylanharper** [sanity-io/sanity#4432]
**Describe the bug**

I upgrade to the latest version of Sanity (`3.9.1`) and notice that the first [Field Group](https://www.sanity.io/docs/field-groups) object no longer displays the corresponding group data or tab within the Sanity Studio.

**To Reproduce**

I have a simple a set of field groups:
```
// Current field groups
groups: [
  {
    title: 'Content',
    name: 'content',
    default: true
  },
  {
    title: 'Upsells',
    name: 'upsells'
  }
],

// Schema fields
fields: [
    {
      title: 'Block Title',
      name: 'blockTitle',
      type: 'string',
      group: 'content'
    },
    {
      title: 'Upsell Image',
      name: 'upsellImage',
      type: 'image',
      description: 'An upsell image showcasing a product or collection',
      group: 'upsells'
    }
  ],
```

The first group `content` no longer displays its tab and data within the array. You can also get rid of the second group set (`upsells`) and it will still not work.

**Expected behavior**

I should see the following field groups at the top of the document: `All fields`, `Content`, `Upsells`. Instead I only see `All fields`, `Upsells` and the first group data no longer displays. You can see this visually in the screenshot below ⬇️

**Screenshots**

<img width="375" alt="Screenshot 2023-04-27 at 12 10 35 PM" src="https://user-images.githubusercontent.com/16216160/234971186-a76b002e-ddc8-42e0-9c48-6229c921cc2a.png">

**Which versions of Sanity are you using?**

Sanity `3.9.1`

**Comment by thierryteyssier** [comment]
Indeed I have the same issue.
And if I have fields without an assigned group, these fields are visible on all groups and not only in "All fields".

**Comment by rylanharper** [comment]
Yes, this is still present in the latest `3.10.0` as well.

**Comment by thobas-dnvgl** [comment]
Ping @bjoerge because you've updated the groups feature in 3.9.1.

**Comment by sebmor** [comment]
+1 Fields without groups are now visible for all groups.

**Comment by thobas-dnvgl** [comment]
On 3.10.1 I still have the issue (first group declared missing from the UI). Let me try to have a reduced test case.

**Comment by bjoerge** [comment]
Thank you @thobas-dnvgl! - I'll take a look. Let's track https://github.com/sanity-io/sanity/issues/4471 separately.

**Comment by rylanharper** [comment]
@thobas-dnvgl seems to be correct. My bad for not checking that earlier! 😬

**Comment by rylanharper** [comment]
Hey @bjoerge I am a bit confused if this was implemented in `3.10.3`? I still have the issue. Or it it scheduled for next release?


https://github.com/sanity-io/sanity/assets/16216160/ce24f72e-59fa-4e02-a2d0-af62e314ad11

**Comment by bjoerge** [comment]
Scheduled for next release

**Comment by rylanharper** [comment]
Sorry to keep reporting this, but this issue is still happening in studio `3.11.0` — can anyone else chime in to see if this issue  is occurring within their studio? My clients and myself still have this issue and its getting in the way of editing larger content fields.. 😞

Again I have a "header" schema (simplified version):
```
groups: [
  {
    title: 'Content',
    name: 'content',
    default: true
  },
  {
    title: 'Upsells',
    name: 'upsells'
  }
],
fieldsets: [
  {
    title: 'Block One',
    name: 'headerBlock1',
    description: 'Content for the first header block',
    options: { collapsible: true }
  }
],
fields: [
  {
    title: 'Block Title',
    name: 'blockTitle1',
    type: 'string',
    fieldset: 'headerBlock1',
    group: 'content'
  },
  {
    title: 'Upsell Image',
    name: 'upsellImage1',
    type: 'image',
    description: 'An upsell image showcasing a product or collection',
    options: { collapsible: true },
    group: 'upsells'
  }
]
```

I should see the following tabs and its corresponding data: `All fields`, `Content`, and `Upsells`. Instead I see only `All fields` and `Upsells` and its corresponding data does not display. Here is a screen record of the issue directly with the above example (ignore the schema warning I have commented out some fields):


https://github.com/sanity-io/sanity/assets/16216160/883612bc-bf98-4380-9b2c-2416324b6543

**Comment by bjoerge** [comment]
Apologies @rylanharper - turns out the fix didn't make it into yesterday's release after all, but I've released v3.11.1 now that includes it.

**Comment by rylanharper** [comment]
It works!!
